src/main/scala/performanceanalysis/LogParserActor.scala

```diff
@@ -20,51 +21,82 @@ class LogParserActor extends Actor with ActorLogging {
+      sender() ! MetricRegistered(metric)
 
     case msg: SubmitLogs =>
-      handleSubmitLogs(msg, metrics, metricToAlertRuleActors)
+      handleSubmitLogs(msg, metrics, alertsByMetric)
+
+    case msg: DeleteAllAlertingRules =>
+      handleDeleteRules(metrics, alertsByMetric, msg)
 
     case msg: RegisterAlertingRule =>
       log.debug("received new alert rule {} in {}", msg, self.path)
 
       metricWithKey(msg.metricKey, metrics) match {
         case None => sender() ! MetricNotFound(msg.componentId, msg.metricKey)
         case Some(metric) =>
+          context.become(normal(metrics, updateAlertsByMetric(alertsByMetric, msg)))
           sender() ! AlertingRuleCreated(msg.componentId, msg.metricKey, msg.rule)
-          context.become(normal(metrics, updateMetricToAlertRuleActors(metricToAlertRuleActors, msg)))
       }
-
   }
 
   private def metricWithKey(metricKey: String, metrics: List[Metric]) =
     metrics.find(_.metricKey == metricKey)
 
-  private def updateMetricToAlertRuleActors(metricToAlertActions: Map[MetricKey, List[ActorRef]], msg: RegisterAlertingRule) = {
+  private def updateAlertsByMetric(alertsByMetric: Map[MetricKey, List[ActorRef]], msg: RegisterAlertingRule) = {
     val alertRuleActor = create(context, msg.rule, msg.componentId, msg.metricKey)
-    val updatedActionActorsForMetric: List[ActorRef] = alertRuleActor :: metricToAlertActions.getOrElse(msg.metricKey, Nil)
-    metricToAlertActions + (msg.metricKey -> updatedActionActorsForMetric)
+    val updatedActionActorsForMetric: List[ActorRef] = alertRuleActor :: alertsByMetric.getOrElse(msg.metricKey, Nil)
+    alertsByMetric + (msg.metricKey -> updatedActionActorsForMetric)
+  }
+
+  private def handleGetAlertRules(alertsByMetric: Map[MetricKey, List[ActorRef]], metricKey: String) = {
+
+    alertsByMetric.get(metricKey) match {
+      case None =>
+        sender() ! MetricNotFound
+      case Some(ruleList) =>
+        context.actorOf(GetAlertsActor.props(ruleList, sender()))
+    }
   }
 
-  private def handleSubmitLogs(msg: SubmitLogs, metrics: List[Metric], metricToAlertActions: Map[MetricKey, List[ActorRef]]) {
+  private def handleDeleteRules(metrics: List[Metric], alertsByMetric: Map[MetricKey, List[ActorRef]], msg: DeleteAllAlertingRules) = {
+    log.debug("received delete request for all rules on metric {}", msg.metricKey)
+
+    alertsByMetric.get(msg.metricKey) match {
+      case None =>
+        sender() ! AlertRulesDeleted(msg.componentId)
+      case Some(ruleList) =>
+        for (ruleActor <- ruleList) {
+          context.stop(ruleActor)
+        }
+
+        context.become(normal(metrics, alertsByMetric - msg.metricKey))
+        sender() ! AlertRulesDeleted(msg.componentId)
+    }
+  }
+
+  private def handleSubmitLogs(msg: SubmitLogs, metrics: List[Metric], alertsByMetric: Map[MetricKey, List[ActorRef]]) {
     log.debug("received {} in {}", msg, self.path)
     for (metric <- metrics) {
```

Comment:
Is it really required `for` in `for`, is not possible to do with single `for` expression